src/main/java/gichat/GiChat.java

```diff
@@ -0,0 +1,228 @@
+import gichat.task.Event;
+import gichat.task.ToDo;
+import gichat.task.Deadline;
+
+/**
+ * Main class for the GiChat bot
+ */
+public class GiChat {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Construct a new GiChat instance with the specified file path
+     *
+     * @param filePath The path to the data file
+     */
+    public GiChat(String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        try {
+            this.tasks = new TaskList(storage.load());
+        } catch (Exception e) {
+            ui.showError("Error loading tasks: " + e.getMessage());
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     *  Main method to start application
+     *
+     * @param args Command line argument
+     */
+    public static void main(String[] args) {
+        new GiChat("data/tasks.txt").run();
+    }
+
+    /**
+     * Runs the main application loop
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+
+        while (!isExit) {
+            try {
+                String fullcomand = ui.readCommand();
+                Command command = Parser.parse(fullcomand);
+                executeCommand(command);
+                isExit = command.isExit();
+            } catch (Exception e) {
+                ui.showError("Error: " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Execute the given command
+     *
+     * @param command The command to execute
+     */
+    public void executeCommand(Command command) {
+        switch (command.getType()) {
+        case BYE:
+            storage.save(tasks.getAllTasks());
+            ui.showGoodbye();
+            break;
+        case LIST:
+            ui.showTasksList(tasks);
+            break;
+        case MARK:
+            handleMarkTask(command.getArguments(), true);
+            break;
+        case UNMARK:
+            handleMarkTask(command.getArguments(), false);
+            break;
+        case TODO:
+            handleAddTodo(command.getArguments());
+            break;
+        case DEADLINE:
+            handleAddDeadline(command.getArguments());
+            break;
+        case EVENT:
+            handleAddEvent(command.getArguments());
+            break;
+        case DELETE:
+            handleDeleteTask(command.getArguments());
+            break;
+        case FIND:
+            handleFindTasks(command.getArguments());
+            break;
+        case UNKNOWN:
+            ui.showError("Erm... you need to give me a valid command...\n" +
+                    "Can list, mark, unmark, todo, deadline, event, delete, find");
+            break;
+        }
+    }
+
+    /**
+     * Handles marking and unmarking of tasks
+     *
+     * @param arguments Task Number
+     * @param markDone True to mark as done, false to mark as undone
+     */
+    private void handleMarkTask(String arguments, boolean markDone) {
```

Comment:
Perhaps the boolean could be named more to sound like a boolean
```suggestion
    private void handleMarkTask(String arguments, boolean isMarked) {
```